config/src/main/java/org/springframework/security/config/annotation/web/configurers/CsrfCustomizer.java

```diff
@@ -0,0 +1,57 @@
+package org.springframework.security.config.annotation.web.configurers;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+import org.springframework.security.web.csrf.CsrfTokenRequestHandler;
+import org.springframework.security.web.csrf.XorCsrfTokenRequestAttributeHandler;
+import org.springframework.util.StringUtils;
+
+import java.util.function.Supplier;
+
+public class CsrfCustomizer {
+	public static Customizer<CsrfConfigurer<HttpSecurity>> spaDefaults() {
+		return (csrf) -> csrf
+				.csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+				.csrfTokenRequestHandler(new SpaCsrfTokenRequestHandler());
+	}
+
+	private static class SpaCsrfTokenRequestHandler implements CsrfTokenRequestHandler {
+		private final CsrfTokenRequestHandler plain = new CsrfTokenRequestAttributeHandler();
+		private final CsrfTokenRequestHandler xor = new XorCsrfTokenRequestAttributeHandler();
+
+		@Override
+		public void handle(HttpServletRequest request, HttpServletResponse response, Supplier<CsrfToken> csrfToken) {
+			/*
+			 * Always use XorCsrfTokenRequestAttributeHandler to provide BREACH protection of
+			 * the CsrfToken when it is rendered in the response body.
+			 */
+			this.xor.handle(request, response, csrfToken);
+			/*
+			 * Render the token value to a cookie by causing the deferred token to be loaded.
+			 */
+			csrfToken.get();
```

Comment:
Instead of calling `csrfToken.get()`, let's please configure  `XorCsrfTokenRequestAttributeHandler#setCsrfRequestAttributeName(null)` in the constructor. This will trigger the token repository to write the value to the cookie.

Note that the reason the documentation uses `csrfToken.get()` is to demonstrate step-by-step what is happening.